### PR TITLE
Add additional metadata to the response

### LINKED ISSUE #7144: Add `numConsumingSegmentsProcessed` and `numConsumingSegmentsMatched` into the query response metadata
Currently we have the following metadata in the query response:
- numSegmentsQueried: number of segments queried by the broker
- numSegmentsProcessed: number of segments processed after the server-side segment pruning
- numSegmentsMatched: number of segments that has records matched
- numConsumingSegmentsQueried: number of consuming segments queried by the broker

We want to also add `numConsumingSegmentsProcessed` and `numConsumingSegmentsMatched` to the query response metadata.

Note: In the `DataTable.MetadataKeys`, we mistakenly use `NUM_CONSUMING_SEGMENTS_PROCESSED` to represent the `numConsumingSegmentsQueried`

### PR DESCRIPTION
**Pull request by ravishankar15**
Fix #7144 
Reference: #9092, #9119

### CODE REVIEW

**Comment by ravishankar15** [comment]
@siddharthteotia @Jackie-Jiang @snleee Can you take a look at the changes ? I have attached the references in the description

**Review by snleee — APPROVED** [review]
LGTM

@Jackie-Jiang @siddharthteotia do you guys want to take a look once more?

File: pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java [hunk 1]
```diff
@@ -670,14 +670,16 @@ private void logBrokerResponse(long requestId, String query, RequestContext requ
     if (_queryLogRateLimiter.tryAcquire() || forceLog(brokerResponse, totalTimeMs)) {
       // Table name might have been changed (with suffix _OFFLINE/_REALTIME appended)
       LOGGER.info("requestId={},table={},timeMs={},docs={}/{},entries={}/{},"
-              + "segments(queried/processed/matched/consuming/unavailable):{}/{}/{}/{}/{},consumingFreshnessTimeMs={},"
+              + "segments(queried/processed/matched/consumingQueried/consumingMatched/consumingProcessed/unavailable):"
```
Comment (Jackie-Jiang) [inline pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java:673]: The ordering is incorrect
```suggestion
              + "segments(queried/processed/matched/consumingQueried/consumingProcessed/consumingMatched/unavailable):"
```

File: pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java [hunk 2]
```diff
@@ -65,13 +68,24 @@ public static void setExecutionStatistics(IntermediateResultsBlock resultsBlock,
...
         numSegmentsMatched++;
       }
+      try {
+        if (operator.getIndexSegment() instanceof MutableSegment) {
+          numConsumingSegmentsProcessed += 1;
+          if (executionStatistics.getNumDocsScanned() > 0) {
+            numConsumingSegmentsMatched++;
+          }
+        }
+      } catch (UnsupportedOperationException ignored) { }
```
Comment (Jackie-Jiang) [inline pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java:78]: This try catch is redundant

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

File: pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java (hunk 2 above)
Comment (ravishankar15) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java:78]: `SelectionCombineOperatorTest.testSelectionLimit0` Should I modify this test to expect an exception as well ? Also while I was trying the query I was getting UnsupportedException in the response but the exception has happened while fetching the metadata. So it gives an impression that the query is not supported which it is.

Let me know if you still feel we want to change this, Thanks

Comment (Jackie-Jiang) [reply pinot-core/src/main/java/org/apache/pinot/core/operator/combine/CombineOperatorUtils.java:78]: I see. All the segment top level operator should have this implemented, or this won't be accurate. Seems it is missing from several operators. We may keep it like this for now, and let's add a TODO to check all operators and make sure they have that method properly implemented in a separate PR.

(nit) We usually use 2 lines even when the block is empty
```suggestion
      } catch (UnsupportedOperationException ignored) {
      }
```

**Comment by snleee** [comment]
Thanks for working on this!

**Pull request merged**